clicommand/agent_start.go

```diff
@@ -362,6 +362,9 @@ var AgentStartCommand = cli.Command{
 		// Setup the any global configuration options
 		HandleGlobalFlags(l, cfg)
 
+		// Remove any config env from the environment to prevent them propagating to bootstrap
+		UnsetConfigFromEnvironment(c)
```

Comment:
I just realised this is going to filter much more than just the `BUILDKITE_AGENT_TOKEN` yeah? For example:

```
env "PATH=$PATH:/usr/local/bin" buildkite-agent start --token xxx
```

Okay for a point release?

Reply:
I hadn't considered this for a point release, only filtering the agent token 🤔

Reply:
No, it will only filter out env used for config for the `start` command.

Reply:
Okay cool! Having re-read that, it totally says it right on the tin.